server.tsx

```diff
@@ -15,195 +16,18 @@ import {
-          url={c.req.url}
-          locale={locale}
-          bodyScript={customBodyScript}
-          headScript={customHeaderScript}
-          favicon={!!favicon}
-          faviconLink={faviconIsUrl ? favicon : undefined}
-          customPages={customPages}
-        />
-      ),
-    );
-  }
-  return c.html(
-    `<!DOCTYPE html>` +
-    (
-      <CustomPage
-        article={page}
-        siteTitle={siteTitle}
-        url={c.req.url}
-        locale={locale}
-        bodyScript={customBodyScript}
-        headScript={customHeaderScript}
-        favicon={!!favicon}
-        faviconLink={faviconIsUrl ? favicon : undefined}
-        customPages={customPages}
-      />
-    ),
-  );
-}
-
-function serveStaticFile(name?: string, folder?: string) {
-  if (!name) {
-    return new Response("Not found", { status: 404 });
-  }
-  try {
-    let file;
-    if (folder) {
-      file = Deno.readFileSync(path.join(folder, name));
-    } else {
-      file = Deno.readFileSync(name);
-    }
-    return new Response(file, {
-      headers: {
-        "content-type":
-          contentType(name.substring(name.lastIndexOf("."))) ||
-          "application/octet-stream",
-      },
-    });
-  } catch {
-    return new Response("Not found", { status: 404 });
-  }
-}
-
-async function getNoArticlesMessage(opts: SmallblogOptions) {
-  const { noArticlesMessage, postsFolder, draftsFolder } = opts;
-
-  if (noArticlesMessage) {
-    return noArticlesMessage;
-  }
-
-  const baseMessage = `<p>You have no articles yet, you can add them by creating a folder <code>${postsFolder}</code> and adding markdown files in it. Don't forget to also add a <code>${draftsFolder}</code> folder for your drafts.</p><p>Read the README for more informations.</p>`;
-
-  const writePermission = await Deno.permissions.query({
-    name: "write",
-    path: ".",
-  });
-
-  if (writePermission.state === "granted") {
-    return (
-      baseMessage +
-      `<p><a class="button" href="/init">Let smallblog do it for you!</a></p>`
-    );
-  }
-
-  return baseMessage;
-}
-
-/**
- * The function to create your blog, you configure your blog with the options
- * and then you just have to write your files
- *
- * @param options The parameters to create the blog.
- * @returns A smallweb/hono app with a fetch method
- */
-export function createSmallblog(options: SmallblogOptions = {}): App {
-  const {
-    postsFolder = "posts/",
-    draftsFolder = "drafts/",
-    pagesFolder = "pages/",
-    favicon,
-    siteTitle = "Smallblog",
-    siteDescription = `The blog: ${siteTitle}`,
-    cacheEnabled = true,
-    indexTitle,
-    indexSubtitle,
-    defaultAuthors = [],
-    locale,
-    customHeaderScript,
-    customBodyScript,
-  } = options;
+export function createServer(params: SmallblogParams) {
+  const server = new Hono();
 
-  const app = new Hono();
+  const postsRoute = path.join("/", params.postsFolder);
+  const draftsRoute = path.join("/", params.draftsFolder);
```

Comment:
Good job! The only thing I don't understand is how the default values are now handled. Have I missed something or did you forget this part?

Reply:
It's there, my bad !